### PR TITLE
fix(platform/gitlab): Strip unicode null characters from markdown

### PR DESCRIPTION
**Pull request by nfm**
Attempting to post a comment with null characters in them fails with a 400 error. See https://github.com/renovatebot/renovate/discussions/19532 for more details.

## Changes

Strip unicode null characters from GitLab markdown, as their presence prevents comments from being posted. bundler appears to generate some error messages that include them.

## Context

https://github.com/renovatebot/renovate/discussions/19532

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/platform/gitlab/index.ts [hunk 1]
```diff
@@ -697,7 +697,9 @@ export function massageMarkdown(input: string): string {
   let desc = input
     .replace(regEx(/Pull Request/g), 'Merge Request')
     .replace(regEx(/PR/g), 'MR')
-    .replace(regEx(/\]\(\.\.\/pull\//g), '](!');
+    .replace(regEx(/\]\(\.\.\/pull\//g), '](!')
+    // Strip unicode null characters as GitLab markdown does not permit them
+    .replace(regEx(/\u0000/g), "");
```
[the remark this answers is not in the document]
Comment (nfm) [reply lib/modules/platform/gitlab/index.ts:702]: I can add this each platforms' versions of `massageMarkdown` 👍 GitHub at least doesn't seem to need it, but I don't think it will do any harm.

I'm probably not the right person to create a function that's shared across all of them though without a few more pointers about where to put it - still getting across the codebase.

[the remark this answers is not in the document]
Comment (rarkins) [reply lib/modules/platform/gitlab/index.ts:702]: I'm ok with merging this as is and deferring the rest.

Right now we don't have a convenient location for massaging all which I'm aware of

**Review by viceice — DISMISSED** [review]

File: lib/modules/platform/gitlab/index.spec.ts [hunk 2]
```diff
@@ -2074,6 +2074,12 @@ These updates have all been created already. Click a checkbox below to force a r
       expect(smartTruncate).toHaveBeenCalledTimes(1);
       expect(smartTruncate).toHaveBeenCalledWith(expect.any(String), 1000000);
     });
+
+    it('strips invalid unicode null characters', async () => {
```
Comment (viceice) [inline lib/modules/platform/gitlab/index.spec.ts:2078]:
```suggestion
    it('strips invalid unicode null characters', () => {
```

**Review by viceice — COMMENTED** [review]
Something is wrong with your solution

**Review by viceice — DISMISSED** [review]

File: lib/modules/platform/gitlab/index.ts (hunk 1 above)
Comment (viceice) [inline lib/modules/platform/gitlab/index.ts:702]: still doesn't work, maybe do a simple string replace instead?

Comment (nfm) [reply lib/modules/platform/gitlab/index.ts:702]: Urgh sorry @viceice, something silly going on here :( The test passes if I run it locally in focus mode (changing `it` to `fit` for my newly added test), but fails when run with other tests in the file.

I did a bit of debugging and the call to `desc = smartTruncate(desc, 25000);` is what's setting `desc` to `undefined` in my test case. If I add a `console.log` to the function body for `smartTruncate`, it's not being reached in my test case.

I think the `jest.mock('../utils/pr-body'); const { smartTruncate } = require('../utils/pr-body');` mocks are not isolated properly between the tests. If I comment out the three tests that do that and run the whole file, my change and newly added test pass.

There is an existing `beforeEach` block that's trying to do something to reset the mocks, but I'm not super familiar with jest, and it doesn't seem to be working in this context.

Do you have any pointers?

Comment (viceice) [reply lib/modules/platform/gitlab/index.ts:702]: move your test above the all other massageMarkdown tests for now

Comment (nfm) [reply lib/modules/platform/gitlab/index.ts:702]: That appears to have worked 👍 Pushed a commit reordering the tests.

File: yarn.lock [hunk 3]
```diff
@@ -2449,7 +2449,6 @@
 
 "@renovate/eslint-plugin@https://github.com/renovatebot/eslint-plugin#v0.0.5":
   version "0.0.5"
-  uid "1923e2e2776549cb25a2c6ea09915ee31bff6d05"
```
Comment (viceice) [inline yarn.lock:2452]: please revert this change

**Review by viceice — COMMENTED** [review]
conflicts

File: yarn.lock (hunk 3 above)
Comment (nfm) [reply yarn.lock]: Not sure how that slipped in - pushed a commit to restore

**Review by viceice — APPROVED** [review]

**Review by rarkins — APPROVED** [review]
